### PR TITLE
feat(yaml): support comments

### PR DESCRIPTION
**Pull request by erezrokah**
Fixes https://github.com/netlify/netlify-cms/issues/3486.
Had to replace `js-yaml` since it doesn't construct an AST.
`yaml` is used by prettier https://github.com/prettier/prettier/pull/4563.

### TODO
- [x] Test with multiline comments
- [x] Fix timestamp handling
- [x] add docs
- [x] Probably fix some e2e tests since due to changes in serialisation

### CODE REVIEW

File: packages/netlify-cms-core/src/formats/yaml.js [hunk 1]
```diff
@@ -1,49 +1,38 @@
-import yaml from 'js-yaml';
-import moment from 'moment';
-import AssetProxy from 'ValueObjects/AssetProxy';
+import yaml from 'yaml';
 import { sortKeys } from './helpers';
 
-const MomentType = new yaml.Type('date', {
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/formats/yaml.js:6]: Couldn't get `moment.isMoment(value)`  and `value instanceof AssetProxy` to evaluate to `true`.
We can add custom types to `yaml`, but looks like it is not required. @erquhart  thoughts?

File: packages/netlify-cms-core/src/formats/__tests__/yaml.spec.js [hunk 2]
```diff
@@ -13,5 +62,73 @@ describe('yaml', () => {
...
+      ).toEqual('# Comment a\na: a\nb:\n  # Comment c\n  c:\n    d: d\n');
+
+      const expected = `# template comment
+template: post
+# title comment
+title: title
+# image comment
+image: /media/netlify.png
+# date comment
+date: "2020-04-02T13:27:48.617Z"
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/formats/__tests__/yaml.spec.js:88]: Only difference I could find between `js-yaml` is that `yaml` uses double quotes and not single quotes (e.g. `'2020-04-02T13:27:48.617Z'`.
Tried https://eemeli.org/yaml/#tag-stringifier-options but that quotes the keys as well.

File: packages/netlify-cms-widget-date/src/DatePreview.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ const DatePreview = ({ value }) => (
 );
 
 DatePreview.propTypes = {
-  value: PropTypes.object,
+  value: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
```
Comment (erezrokah) [inline packages/netlify-cms-widget-date/src/DatePreview.js:10]: unrelated prop warning fix

File: packages/netlify-cms-widget-datetime/src/DateTimePreview.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ const DatePreview = ({ value }) => (
 );
 
 DatePreview.propTypes = {
-  value: PropTypes.object,
+  value: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
```
Comment (erezrokah) [inline packages/netlify-cms-widget-datetime/src/DateTimePreview.js:10]: unrelated prop warning fix

File: packages/netlify-cms-widget-markdown/src/MarkdownControl/RawEditor.js [hunk 5]
```diff
@@ -125,7 +125,6 @@ export default class RawEditor extends React.Component {
               onCut={this.handleCut}
               onCopy={this.handleCopy}
               ref={this.processRef}
-              t={t}
```
Comment (erezrokah) [inline packages/netlify-cms-widget-markdown/src/MarkdownControl/RawEditor.js:128]: unrelated prop warning fix

File: packages/netlify-cms-core/src/formats/yaml.js (hunk 1 above)
Comment (erquhart) [reply packages/netlify-cms-core/src/formats/yaml.js:6]: The date widget used to pass a moment instance if no format was configured, but that doesn't seem possible anymore:
https://github.com/netlify/netlify-cms/blob/master/packages/netlify-cms-widget-date/src/DateControl.js#L94

Maybe something similar happened with `AssetProxy`, but ensuring that may be more complex than just checking the change handler for the image widget. If you feel confident that an instance of `AssetProxy` won't end up as a field value in the redux store, we should be good to remove the related logic here.

Comment (erezrokah) [reply packages/netlify-cms-core/src/formats/yaml.js:6]: Looks like the image widget(s) pass strings/arrays of strings:
https://github.com/netlify/netlify-cms/blob/2fe9ab72f38a1659011979069cfe196a2c35cff3/packages/netlify-cms-widget-file/src/withFileControl.js#L226
https://github.com/netlify/netlify-cms/blob/2fe9ab72f38a1659011979069cfe196a2c35cff3/packages/netlify-cms-core/src/actions/mediaLibrary.ts#L112

**Pull request merged**